feat: Add GitHub integration fields to Issue model

satvikpathak opened the pull request:
This PR adds comprehensive test coverage for the GitHub integration fields on the Issue model and improves data integrity by using `PositiveIntegerField` for comment counts.

## GitHub Integration Fields Added

This PR works with the following 4 GitHub integration fields that were added to the Issue model:

* **`github_comment_count`** (`PositiveIntegerField`) - Number of comments on the GitHub issue
* **`github_state`** (`CharField`) - Current state of the GitHub issue (open/closed) as reported by GitHub API
* **`github_fetch_status`** (`BooleanField`) - Whether GitHub data was successfully fetched from the API
* **`github_last_fetched_at`** (`DateTimeField`) - Timestamp of when GitHub data was last fetched (tracks freshness of GitHub data)

## Changes Made

### Model Improvements
* Changed `github_comment_count` from `IntegerField` to `PositiveIntegerField` to prevent negative values and ensure data integrity.
* Updated migration (0260) to use `PositiveIntegerField` to match model definition.

### Test Coverage Added
* **`test_issue_github_last_fetched_at`** - Verifies timestamp tracking for GitHub data fetches:
    * Default value is `None` for new issues.
    * Timestamp is set when GitHub data is successfully fetched.
    * Timestamp is updated on subsequent fetches.
* **`test_issue_github_state_invalid_choice`** - Ensures invalid state values are rejected:
    * Uses Django's `full_clean()` validation.
    * Expects `ValidationError` for invalid choices.
* **`test_issue_github_comment_count_rejects_negative`** (updated) - Verifies negative values are rejected:
    * Previously documented that negative values were allowed.
    * Now properly tests that `PositiveIntegerField` rejects negative values.

satvikpathak committed:
feat: Add GitHub integration fields to Issue model

- Add github_comment_count field to track number of comments
- Add github_state field with open/closed choices
- Add github_fetch_status field to indicate successful data fetch
- Update _bug.html template with GitHub information display
- Add GitHub state badges (open/closed) with Tailwind CSS
- Add comment count display with proper pluralization
- Include 'Data unavailable' state for failed fetches
- Full dark mode support for GitHub UI components
- Migration 0260 adds three new fields to Issue model

Fixes #5053

sidd190 commented on website/migrations/0260_add_github_fields_to_issue.py:
```diff
@@ -0,0 +1,33 @@
+
+class Migration(migrations.Migration):
+    dependencies = [
+        ("website", "0259_add_domain_slug"),
```
Kindly address the comment here about non existent migration number, this is an issue which will throw error on applying migrations

sidd190 reviewed (changes requested):
### github_state field: remove null=True, use default="" (avoid NULL vs empty-string ambiguity)

Problem: In the migration and model, github_state is declared with null=True and blank=True. For CharFields Django best practice is to avoid null=True — prefer blank=True, default="". Using null=True creates two “empty” states (NULL and "") and complicates queries. CI also flagged this.

suggested Fixes:

Update migration’s AddField for github_state to remove null=True and set default="".
Update website/models.py Issue field definition to match.

Please take a look at this and the other comment. Looks good otherwise.

satvikpathak committed:
Fix: Update github_state field to use default='' instead of null=True

- Remove null=True from github_state CharField (Django best practice)
- Add default='' to avoid NULL vs empty string ambiguity
- Fix migration dependency to depend on 0258 instead of unpushed 0259

Pritz395 reviewed (changes requested):
Just need to update tests and you're all good. The model uses default="" for github_state (no null=True), but 5 tests still assert None:

Line 118: test_issue_created_with_default_github_fields
Line 187: test_issue_github_fetch_failed
Line 200: test_issue_update_github_fields
Line 290: test_multiple_issues_with_different_github_states
Line 303: test_issue_github_state_null_allowed

With default="", new issues have github_state="", not None. Update these assertions to assertEqual(issue.github_state, "") to match the model.

satvikpathak committed:
Fix test assertions for github_state field to use empty string instead of None

satvikpathak committed:
Add comprehensive tests for GitHub integration fields and improve data integrity

- Changed github_comment_count from IntegerField to PositiveIntegerField to prevent negative values
- Added test_issue_github_last_fetched_at to verify timestamp tracking for GitHub data fetches
- Added test_issue_github_state_invalid_choice to ensure invalid state values are rejected
- Updated test_issue_github_comment_count_rejects_negative to verify negative values are now rejected

These changes improve test coverage for the GitHub integration feature and ensure data integrity.

satvikpathak committed:
Fix migration: use PositiveIntegerField for github_comment_count

Changed github_comment_count from IntegerField to PositiveIntegerField in the migration to match the model definition and prevent negative comment counts.

DonnieBLT commented:
is there a section where the data is added?  I only see the new fields and not the logic to add the data
